bench/query.go

```diff
@@ -132,6 +132,90 @@ func (q *QueryGenerator) Random(maxN, depth, maxargs int, idmin, idmax uint64) *
 	}
 }
 
+// RandomRangeQuery returns a randomly generated either sum/range query.
+func (q *QueryGenerator) RandomRangeQuery(depth, maxargs int, frame, field string, idmin, idmax uint64) *pql.Call {
+	val := q.R.Intn(5)
+	switch val {
+	case 1:
+		return q.RandomSum(depth, maxargs, frame, field, idmin, idmax)
+	default:
+		return q.RandomRange(maxargs, field, idmin, idmax)
+	}
+}
+
+func (q *QueryGenerator) RandomRange(numArg int, field string, idmin, idmax uint64) *pql.Call {
+	call := q.R.Intn(4)
+	if call == 0 {
+		return q.RangeCall(field, idmin, idmax)
+	}
+	calls := make([]*pql.Call, numArg)
+	for i := 0; i < numArg; i++ {
+		calls[i] = q.RangeCall(field, idmin, idmax)
+	}
+
+	switch call {
+	case 1:
+		return Difference(calls...)
+	case 2:
+		return Intersect(calls...)
+	case 3:
+		return Union(calls...)
+	}
+
+	return nil
+}
+
+func (q *QueryGenerator) RangeCall(field string, idmin, idmax uint64) *pql.Call {
+
+	var operations = []pql.Token{pql.GT, pql.LT, pql.GTE, pql.LTE}
+	opIndex := q.R.Intn(len(operations))
+	frameIdx := q.R.Intn(len(q.Frames))
+
+	condition := pql.Condition{operations[opIndex], q.R.Intn(int(idmax - idmin))}
+	return &pql.Call{
+		Name: "Range",
+		Args: map[string]interface{}{
+			"frame": q.Frames[frameIdx],
+			field:   &condition,
+		},
+	}
+}
+
+// RandomSum returns a randomly generated sum query.
+func (q *QueryGenerator) RandomSum(depth, maxargs int, frame, field string, idmin, idmax uint64) *pql.Call {
+	frameIdx := q.R.Intn(len(q.Frames))
+	val := q.R.Intn(5)
+	switch val {
+	case 0:
+		return &pql.Call{
```

Comment:
these three blocks seem to contain very similar code. perhaps make variable `c := &pql.Call{ ... }` before the switch, and just adjust the differences in the case block.